### PR TITLE
Add custom parent-child relationships example

### PR DESCRIPTION
**Pull request by dat-a-man**
### Description

Added example to load parent table keys into child table records.

issue- #1589

### CODE REVIEW

File: docs/examples/parent_child_relationship/parent_child_relationship.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+"""
+---
+title: Load parent table records into child table
+description: Learn how to integrate custom parent keys into child records
+keywords: [parent child relationship, parent key]
+---
+
+This example demonstrates handling data with parent-child relationships using the `dlt` library. This process
+is about integrating specific fields (e.g., primary, foreign keys) from a parent record into each child record,
+ensuring that we can use self-defined parent keys in the child table.
```
Comment (VioletM) [inline docs/examples/parent_child_relationship/parent_child_relationship.py:10]:
```suggestion
This example demonstrates handling data with parent-child relationships using the `dlt` library. You learn how to integrate specific fields (e.g., primary, foreign keys) from a parent record into each child record.
```

File: docs/examples/parent_child_relationship/parent_child_relationship.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+In this example, we'll explore how to:
+
+- Integrate a custom `parent_id` into each child record.
```
Comment (VioletM) [inline docs/examples/parent_child_relationship/parent_child_relationship.py:14]: too much chat-gpt 

```suggestion
- Add `parent_id` into each child record using `add_parent_id` function
- Use the [`add_map` function](https://dlthub.com/docs/api_reference/extract/resource#add_map) to apply this
custom logic to every record in the dataset
```

**Review by VioletM — CHANGES_REQUESTED** [review]

File: docs/examples/parent_child_relationship/parent_child_relationship.py (hunk 1 above)
Comment (dat-a-man) [reply docs/examples/parent_child_relationship/parent_child_relationship.py:10]: done

File: docs/examples/parent_child_relationship/parent_child_relationship.py (hunk 2 above)
Comment (dat-a-man) [reply docs/examples/parent_child_relationship/parent_child_relationship.py:14]: done

**Review by VioletM — APPROVED** [review]

**Pull request merged**